system/Common.php

```diff
@@ -441,7 +441,7 @@ function esc($data, string $context = 'html', ?string $encoding = null)
                 $escaper = new Escaper($encoding);
             }
 
-            if ($encoding && $escaper->getEncoding() !== $encoding) {
+            if ((string) $encoding !== '' && $escaper->getEncoding() !== $encoding) {
```

Comment:
Why not `!== null`

Reply:
If the variable is an empty string, the exception will be thrown later inside the `Escaper` class.